src/libstore/include/nix/store/globals.hh

```diff
@@ -484,6 +484,35 @@ public:
         "id-count",
         "The number of UIDs/GIDs to use for dynamic ID allocation."};
 
+    Setting<StringSet> supplementaryGroups{
+        this, {"kvm"}, "supplementary-groups",
+        R"(
+          A list of supplementary groups to be added to build users when dynamic
+          UID/GID allocation [`auto-allocate-uids`](#conf-auto-allocate-uids) is
+          enabled.
+
+          By default, each listed group is mapped into the build sandbox using its
+          host system GID. You can override the mapped GID by appending `:GID` to
+          the group name. If a group does not exist on the host (i.e., has no
+          assigned GID), it is silently ignored.
+
+          Example:
+
+          ```nix
+          extra-supplementary-groups = nixbld users:10100
```

Comment:
This should probably be a JSON option, similar to https://github.com/NixOS/nix/pull/13315#discussion_r2139529037.

E.g.
```
extra-supplementary-groups = [ {"group": "nixbld"}, {"group": "users", "gid": 10100} ]
```